futures/tests/futures_unordered.rs

```diff
@@ -131,38 +129,10 @@ fn iter_mut_len() {
 
 #[test]
 fn futures_not_moved_after_poll() {
-    struct DontMoveMe {
-        this: *const DontMoveMe,
-        _pinned: Pinned,
-    }
-
-    impl Future for DontMoveMe {
-        type Output = ();
-        fn poll(self: PinMut<Self>, cx: &mut task::Context)
-            -> Poll<Self::Output>
-        {
-            let cur_this = &*self as *const DontMoveMe;
-            if self.this == ptr::null() {
-                // First time being polled.
-                cx.local_waker().wake();
-                unsafe { PinMut::get_mut_unchecked(self).this = cur_this };
-                Poll::Pending
-            } else {
-                assert_eq!(self.this, cur_this, "Future moved between poll calls");
-                Poll::Ready(())
-            }
-        }
-    }
-
-    impl Drop for DontMoveMe {
-        fn drop(&mut self) {
-            let cur_this = &*self as *const DontMoveMe;
-            assert_eq!(self.this, cur_this, "Future moved before drop");
-        }
-    }
-
-    let dmm = || DontMoveMe { this: ptr::null(), _pinned: Pinned };
-    let mut stream = block_on_stream(futures_unordered(vec![dmm(), dmm(), dmm()]));
+    // Future that will be ready after being polled twice,
+    // asserting that it does not move.
+    let fut = future::ready(()).pending_once().assert_unmoved();
+    let mut stream = block_on_stream(futures_unordered(vec![fut; 3]));
     assert_eq!(Some(()), stream.next());
     assert_eq!(Some(()), stream.next());
     assert_eq!(Some(()), stream.next());
```

Comment:
Should probably use [`assert_stream_next!` and `assert_stream_done!`](https://github.com/rust-lang-nursery/futures-rs/blob/6f60ab33d0cb3c148ce34af49a8af38f5487dcd4/futures-test/src/assert.rs#L61-L63)

Reply:
Done.